### PR TITLE
Add tests for NewFilteredDynamicSharedInformerFactory

### LINKED ISSUE #71714: NewFilteredDynamicSharedInformerFactory ignores namespace parameter
**What happened**:

The `NewFilteredDynamicSharedInformerFactory` constructor takes a `namespace` argument which is meant to populate a member and be passed along to new informers constructed in `ForResource`. However, `namespace` is ignored and the `namespace` member is initialized to `metav1.NamespaceAll` instead.

**What you expected to happen**:

I exepected `NewFilteredDynamicSharedInformerFactory` to capture the `namespace` argument in `dynamicSharedInformerFactory.namespace`, and subsequently pass it on to `NewFilteredDynamicInformer` calls in `ForResource`. 

**How to reproduce it (as minimally and precisely as possible)**:

Construct a `FilteredDynamicSharedInformerFactory` scoped to a specific namespace. Create an informer using `ForResource` and perform a `List`. Results will span namespaces.

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
v1.13.0
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

/kind bug

### PR DESCRIPTION
**Pull request by wwitzel3**
Signed-off-by: Wayne Witzel III <witzelw@vmware.com>

**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:
Adds tests to the fix for `NewFilteredDynamicSharedInformerFactory`.

**Which issue(s) this PR fixes**:

Adds tests for #71714 which was fixed in #76776

**Special notes for your reviewer**:

NONE

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by lavalamp** [comment]
/ok-to-test

is this not covered by a test?

/assign @caesarxuchao

File: staging/src/k8s.io/client-go/dynamic/dynamicinformer/informer_test.go [hunk 1]
```diff
@@ -32,6 +32,104 @@ import (
...
+		handler     func(rcvCh chan<- *unstructured.Unstructured) *cache.ResourceEventHandlerFuncs
+	}{
+		// scenario 1
+		{
+			name:     "scenario 1: test adding an object in different namespace should not trigger AddFunc",
+			informNS: "ns-bar",
+			ns:       "ns-foo",
+			gvr:      schema.GroupVersionResource{Group: "extensions", Version: "v1beta1", Resource: "deployments"},
+			trigger: func(gvr schema.GroupVersionResource, ns string, fakeClient *fake.FakeDynamicClient, _ *unstructured.Unstructured) *unstructured.Unstructured {
+				testObject := newUnstructured("extensions/v1beta1", "Deployment", "ns-foo", "name-foo")
```
Comment (yue9944882) [inline staging/src/k8s.io/client-go/dynamic/dynamicinformer/informer_test.go]: extensions/v1beta1 -> apps/v1?

File: staging/src/k8s.io/client-go/dynamic/dynamicinformer/informer_test.go [hunk 2]
```diff
@@ -32,6 +32,104 @@ import (
...
+			},
+		},
+		// scenario 2
+		{
+			name:     "scenario 2: test adding an object should trigger AddFunc",
+			informNS: "ns-foo",
+			ns:       "ns-foo",
+			gvr:      schema.GroupVersionResource{Group: "extensions", Version: "v1beta1", Resource: "deployments"},
+			trigger: func(gvr schema.GroupVersionResource, ns string, fakeClient *fake.FakeDynamicClient, _ *unstructured.Unstructured) *unstructured.Unstructured {
+				testObject := newUnstructured("extensions/v1beta1", "Deployment", "ns-foo", "name-foo")
```
Comment (yue9944882) [inline staging/src/k8s.io/client-go/dynamic/dynamicinformer/informer_test.go]: ditto

File: staging/src/k8s.io/client-go/dynamic/dynamicinformer/informer_test.go [hunk 3]
```diff
@@ -32,6 +32,104 @@ import (
...
+					},
+				}
+			},
+		},
+		// scenario 2
+		{
+			name:     "scenario 2: test adding an object should trigger AddFunc",
+			informNS: "ns-foo",
+			ns:       "ns-foo",
+			gvr:      schema.GroupVersionResource{Group: "extensions", Version: "v1beta1", Resource: "deployments"},
```
Comment (yue9944882) [inline staging/src/k8s.io/client-go/dynamic/dynamicinformer/informer_test.go]: here too

File: staging/src/k8s.io/client-go/dynamic/dynamicinformer/informer_test.go [hunk 4]
```diff
@@ -32,6 +32,104 @@ import (
...
+			}
+
+			testObject := ts.trigger(ts.gvr, ts.ns, fakeClient, ts.existingObj)
+			select {
+			case objFromInformer := <-informerReciveObjectCh:
+				if ts.ns != ts.informNS {
+					t.Errorf("informer received an object for namespace %s when watching namespace %s", ts.ns, ts.informNS)
+				}
+				if !equality.Semantic.DeepEqual(testObject, objFromInformer) {
+					t.Fatalf("%v", diff.ObjectDiff(testObject, objFromInformer))
```
Comment (yue9944882) [inline staging/src/k8s.io/client-go/dynamic/dynamicinformer/informer_test.go]: nit: what's difference between `t.Errorf` and `t.Fatalf` here?

Comment (wwitzel3) [reply staging/src/k8s.io/client-go/dynamic/dynamicinformer/informer_test.go]: Honestly, I don't know why `t.Fatalf` is used there. I copied the structure from the other dynamicinformer test and that is what is does. I am happy to change it either way.

**Review by caesarxuchao — COMMENTED** [review]
A nit, otherwise lgtm.

**Comment by wwitzel3** [comment]
#77945 which was merged and fixes this same thing did not provide any tests around this change I'd like to get these tests merged in.

**Comment by yliaog** [comment]
i guess you don't need the change for informer.go, it's good to have the added test merged

**Review by yue9944882 — APPROVED** [review]
/lgtm

**Comment by wwitzel3** [comment]
Anything else that needs to happen to get this approved? Cheers.

**Comment by wwitzel3** [comment]
/cc @deads2k @yliaog 

Still waiting on approval for these tests.

**Comment by lavalamp** [comment]
...has this just been waiting for an approve for 3 months? Please feel free to ping us more aggressively...

/approve

Given that @yliaog already lgtm'd.

**Comment by wwitzel3** [comment]
@lavalamp I had to fix the signature of a failing test, so it will need another review / approval. Cheers.

**Comment by lavalamp** [comment]
I asked in the #testing slack channel if pull-verify has got problems right now.

**Pull request merged**